addons/utm/models/utm.py

```diff
@@ -26,6 +27,16 @@ class UtmMedium(models.Model):
     name = fields.Char(string='Medium Name', required=True)
     active = fields.Boolean(default=True)
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_utm_medium_email(self):
+        utm_medium_email = self.env.ref('utm.utm_medium_email', raise_if_not_found=False)
```

Comment:
Seems referral uses utm.utm_medium_direct, utm.utm_medium_facebook, utm.utm_medium_twitter, utm.utm_medium_linkedin . Shoudln't we also consider tham as "master data", even if a bit less master than email or sms ? Seems logical to keep those as they are often used maybe ?

Reply:
I actually added some code bits to avoid crashes if those are removed.
They are now used but "nice to have".

I would rather still allow their deletion (Odoo style, don't restrict the user, etc.)